### ISSUE TITLE
default_model.bin.gz does not work under non-ascii path

### ISSUE THREAD

**Issue by kaorahi** [lightvector/KataGo#518]
I am quite unfamiliar with Windows. But from @hebaeba's suggestion, I noticed the following error.

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a> .\katago\katago-eigen.exe gtp
Uncaught exception: -model MODELFILENAME.bin.gz was not specified to tell KataGo where to find the neural net model, and default was not found at /default_model.bin.gz
~~~

I believe default_model.bin.gz exists.

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a> ls .\katago\default_model.bin.gz


    ディレクトリ: C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a\katago


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       2021/07/04     12:29       36948927 default_model.bin.gz
~~~

The error disappears if I rename the folder `碁` to `a`.

~~~
PS C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_210704a> .\katago\katago-eigen.exe gtp
KataGo v1.9.1
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Loaded config C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_210704a\katago/default_gtp.cfg
Loaded model C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_210704a\katago/default_model.bin.gz
Model name: g170-b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~

**Comment by fuhaoda** [comment]
I just tried it in Mac OS. It seems that there is no such issue with either using 碁 or other non-ascii as path. Could you please provide a little bit more information on which version of Windows that you are using, so that we can reproduce the results? It seems an operation system dependent issue. https://stackoverflow.com/questions/62357376/windows-non-ascii-file-paths

Also, could you please try whether double quote your path works, such as using "C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a\katago\default_model.bin.gz"?

**Comment by kaorahi** [comment]
thx for the reply. additional info:

* Windows 10 Home 64bit
* Similar issues are also observed for `default_gtp.cfg`.
* I refer to "UI and Configs" in the release notes of v1.3.4 for the intended behavior of these `default_*` files. https://github.com/lightvector/KataGo/releases/tag/v1.3.4
* I can avoid the issue by explicitly giving their paths as follows.

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a> .\katago\katago-eigen.exe gtp
Uncaught exception: -model MODELFILENAME.bin.gz was not specified to tell KataGo where to find the neural net model, and default was not found at /default_model.bin.gz

PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a> .\katago\katago-eigen.exe gtp -config .\katago\default_gtp.cfg -model .\katago\default_model.bin.gz
KataGo v1.9.1
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Loaded config .\katago\default_gtp.cfg
Loaded model .\katago\default_model.bin.gz
Model name: g170-b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~

Your last comment on "double quote" does not make sense for me. I have no argument to double quote except for `gtp`. Anyway, I get the same result.

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_210704a> .\katago\katago-eigen.exe "gtp"
Uncaught exception: -model MODELFILENAME.bin.gz was not specified to tell KataGo where to find the neural net model, and default was not found at /default_model.bin.gz
~~~

**Comment by kaorahi** [comment]
@lightvector, would you reopen this issue?

It seems that KataGo v1.10.0 still has the same issue (thx > @hebaeba).

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a> .\katago\katago-eigen.exe gtp
Uncaught exception: -model MODELFILENAME.bin.gz was not specified to tell KataGo where to find the neural net model, and default was not found at /default_model.bin.gz
~~~

There exists `default_model.txt.gz` actually. (I use `.txt.gz` this time as I cannot find `.bin.gz` of the best 15b network in kata1.)

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a> ls .\katago\default_model.txt.gz


    ディレクトリ: C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----       2020/11/28     20:51       48038914 default_model.txt.gz
~~~

KataGo works if `-model ...` is given explicitly.

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a> .\katago\katago-eigen.exe gtp -config katago/default_gtp.cfg -model katago/default_model.txt.gz
KataGo v1.10.0
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Initializing board with boardXSize 19 boardYSize 19
Loaded config katago/default_gtp.cfg
Loaded model katago/default_model.txt.gz
Model name: b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~

KataGo also works if I rename the folder `碁` to `a`.

~~~
PS C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_211211a> .\katago\katago-eigen.exe gtp
KataGo v1.10.0
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Initializing board with boardXSize 19 boardYSize 19
Loaded config C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_211211a\katago/default_gtp.cfg
Loaded model C:\Users\z\Desktop\a\Lizzie-patched-0.7.4_211211a\katago/default_model.txt.gz
Model name: b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~

**Comment by lightvector** [comment]
Reopening, if you are interested in helping investigate, I can work on some test exes in the next few weeks to try to figure out why this is still happening. In particular, internally what string character encoding is being used by the time the path gets to KataGo's code.

**Comment by kaorahi** [comment]
Thanks for the reply. I am willing to do the tests unless they need some installation, network connection, etc. (I am not a Windows user and have no Windows machine that I can use freely.)

**Comment by lightvector** [comment]
Okay, https://github.com/lightvector/KataGo/pull/594 is a branch that attempts to fix this.

Here are two test exes you can try, compiled from that branch. Hopefully this works now? :)
[katago-v1.10.0-eigen-windows-x64-test.zip](https://github.com/lightvector/KataGo/files/7803405/katago-v1.10.0-eigen-windows-x64-test.zip)
[katago-v1.10.0-opencl-windows-x64-test.zip](https://github.com/lightvector/KataGo/files/7803406/katago-v1.10.0-opencl-windows-x64-test.zip)

**Comment by kaorahi** [comment]
Both exes work fine under the non-ascii path `碁`. Thanks for the fix!

eigen:

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a> .\katago\katago-test220103a-eigen.exe gtp
KataGo v1.10.0
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Initializing board with boardXSize 19 boardYSize 19
Loaded config C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago/default_gtp.cfg
Loaded model C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago/default_model.txt.gz
Model name: b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~

opencl:

~~~
PS C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a> .\katago\katago-test220103a-opencl.exe gtp
KataGo v1.10.0
Using TrompTaylor rules initially, unless GTP/GUI overrides this
Creating context for OpenCL Platform: Intel(R) OpenCL (Intel(R) Corporation) (OpenCL 2.1 )
...(snip)...
Tuning 100/106 ...
Done tuning
------------------------------------------------------
Done tuning, saved results to C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago/KataGoData/opencltuning/tune8_gpuIntelRUHDGraphics620_x19_y19_c192_mv8.txt
Initializing board with boardXSize 19 boardYSize 19
Loaded config C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago/default_gtp.cfg
Loaded model C:\Users\z\Desktop\碁\Lizzie-patched-0.7.4_211211a\katago/default_model.txt.gz
Model name: b15c192-s1672170752-d466197061
GTP ready, beginning main protocol loop
~~~
